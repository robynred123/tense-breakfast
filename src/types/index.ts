export interface ReducerState {
  therapists: TherapistInfo[];
  filteredTherapists: TherapistInfo[];
  filterOptions: FilterOptions;
  mobileFilter: boolean;
  error: string | null;
}

export interface TherapistInfo {
  id: string;
  firstName: string;
  lastName: string;
  appointment_types: AppointmentType[];
  appointment_mediums: AppointmentMedium[];
  specialisms: Specialisms[];
}

export interface DateRange {
  start: string | null;
  end: string | null;
}

export interface AvailabilityData {
  id: string;
  datetime: string;
}

export interface Availabilities {
  [therapistId: string]: AvailabilityData[];
}

export type Type = 'date' | 'type' | 'specialism';

export type AppointmentType = 'one_off' | 'consultation';
export type AppointmentMedium = 'phone' | 'video';
export type Specialisms = string;

export type FilterOptions = {
  appointmentType: AppointmentType[];
  dateRange: DateRange;
  specialisms: Specialisms[];
};

export type BookingOptions = {
  therapistId: string;
  appointmentType: AppointmentType | null;
  appointmentMedium: AppointmentMedium | null;
  appointmentDate: string;
};
