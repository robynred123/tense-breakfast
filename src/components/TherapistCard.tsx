import { Typography } from '@mui/material';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import { DARK_GREY, GREY } from '../constants/colours';
import { TherapistInfo } from '../types';

export const TherapistCard = (props: TherapistInfo) => {
  const therapist = props;
  console.log(therapist);
  return (
    <div style={{ border: 'double', borderColor: DARK_GREY }}>
      <Box
        sx={{
          flexGrow: 1,
          display: 'flex',
          backgroundColor: GREY,
          alignContent: 'center',
          padding: '2%',
        }}
      >
        <Grid container>
          <Grid item xs={2}></Grid>
          <Grid item xs={10} sx={{ padding: '2%', alignContent: 'center', flexGrow: 1 }}>
            <div style={{ borderWidth: '10px', borderColor: 'black' }}>
              <Typography variant='h4' fontFamily={'lato, sans-serif'}>
                {therapist.firstName} {therapist.lastName}
              </Typography>
            </div>
          </Grid>
        </Grid>
      </Box>
    </div>
  );
};
