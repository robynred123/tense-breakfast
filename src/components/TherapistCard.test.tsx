/* eslint-disable testing-library/await-async-query */
import React from 'react';
import renderer from 'react-test-renderer';
import { TherapistCard } from './TherapistCard';
import { TherapistInfo } from '../types';

describe('<TherapistCard />', () => {
  it('Should render the Therapist Card component successfully', () => {
    const testTherapist: TherapistInfo = {
      id: '063eb398-45f7-4a9c-a5d3-e95d14c05c44',
      firstName: 'Albert',
      lastName: 'Einstein',
      appointment_types: ['one_off'],
      appointment_mediums: ['phone'],
      specialisms: ['alcoholism'],
    };
    const view = renderer.create(<TherapistCard {...testTherapist} />);
    expect(view.toJSON()).toMatchSnapshot();
    expect(view.root.findByType('svg').props['data-testid']).toEqual('AccountCircleIcon');
    expect(view.root.findByType('h4').props.children).toEqual(['Albert', ' ', 'Einstein']);
  });
});
