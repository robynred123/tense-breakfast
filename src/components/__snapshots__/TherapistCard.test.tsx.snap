// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TherapistCard /> Should render the Therapist Card component successfully 1`] = `
<div
  style={
    Object {
      "border": "double",
      "borderColor": "#374151",
    }
  }
>
  <div
    className="MuiBox-root css-1caqi7t"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccountCircleIcon"
          focusable="false"
          style={
            Object {
              "color": "#35D0BA",
              "height": "100%",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-uvmdr-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
        >
          <div
            style={
              Object {
                "borderColor": "black",
                "borderWidth": "10px",
              }
            }
          >
            <h4
              className="MuiTypography-root MuiTypography-h4 css-od8ppe-MuiTypography-root"
            >
              Albert
               
              Einstein
            </h4>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <button
            className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-17eaqzp-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Book
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
